src/Dashboard/Provider.php

```diff
@@ -1625,376 +1627,39 @@ public static function formatMonthyearDates(string $monthyear): array
-                'link'       => 'AND',
-                'field'      => self::getSearchOptionID($table, 'requesttypes_id', 'glpi_requesttypes'), // request type
-                'searchtype' => 'equals',
-                'value'      => (int) $apply_filters['requesttype']
-            ];
-        }
-
-        if (
-            $DB->fieldExists($table, 'locations_id')
-            && isset($apply_filters['location'])
-            && (int) $apply_filters['location'] > 0
-        ) {
-            $s_criteria['criteria'][] = [
-                'link'       => 'AND',
-                'field'      => self::getSearchOptionID($table, 'locations_id', 'glpi_locations'), // location
-                'searchtype' => 'equals',
-                'value'      => (int) $apply_filters['location']
-            ];
-        }
-
-        if (
-            $DB->fieldExists($table, 'manufacturers_id')
-            && isset($apply_filters['manufacturer'])
-            && (int) $apply_filters['manufacturer'] > 0
-        ) {
-            $s_criteria['criteria'][] = [
-                'link'       => 'AND',
-                'field'      => self::getSearchOptionID($table, 'manufacturers_id', 'glpi_manufacturers'), // manufacturer
-                'searchtype' => 'equals',
-                'value'      => (int) $apply_filters['manufacturer']
-            ];
-        }
-
-        if (isset($apply_filters['group_tech'])) {
-            $groups_id = null;
-            if ((int) $apply_filters['group_tech'] > 0) {
-                $groups_id =  (int) $apply_filters['group_tech'];
-            } else if ($apply_filters['group_tech'] == 'mygroups') {
-                $groups_id =  'mygroups';
-            }
-
-            if ($groups_id != null) {
-                if ($DB->fieldExists($table, 'groups_id_tech')) {
-                    $s_criteria['criteria'][] = [
-                        'link'       => 'AND',
-                        'field'      => self::getSearchOptionID($table, 'groups_id_tech', 'glpi_groups'), // group tech
-                        'searchtype' => 'equals',
-                        'value'      => $groups_id
-                    ];
-                } else if (
-                    in_array($table, [
-                        Ticket::getTable(),
-                        Change::getTable(),
-                        Problem::getTable(),
-                    ])
-                ) {
-                    $s_criteria['criteria'][] = [
-                        'link'       => 'AND',
-                        'field'      => 8, // group tech
-                        'searchtype' => 'equals',
-                        'value'      => $groups_id
-                    ];
-                }
+        foreach ($filters as $filter) {
+            $filter_criteria = $filter::getSearchCriteria($DBread, $table, $apply_filters);
+            if (!empty($filter_criteria)) {
+                $s_criteria = array_merge($s_criteria, $filter_criteria);
```

Comment:
Search criteria are not working because they is one unneeded array level:

![image](https://github.com/glpi-project/glpi/assets/42734840/40bee213-a73f-4abe-bbd6-9a31526d3c93)

Can be fixed like this:

```suggestion
                array_push($s_criteria, ...$filter_criteria);
```

After fix:

![image](https://github.com/glpi-project/glpi/assets/42734840/af4e29b9-2adc-4c55-bcf8-a5c3eae2e203)